src/SchemaDesign/SchemaDesigner.tsx

```diff
@@ -0,0 +1,418 @@
+    const updatedQuestions = reorder(stepToAmmend.questions, result.source.index, result.destination.index)
+
+    setSteps(steps.map((step) => (step.reference === selectedStep ? { ...step, questions: updatedQuestions } : step)))
+  }
+
+  const deleteStep = (step: string) => {
+    const updatedSteps = steps.filter((stepToRemove) => stepToRemove === step)
+    setSteps(updatedSteps)
+  }
+
+  const getQuestionIcon = (type: string) => {
+    if (type === 'date') {
+      return <CalendarMonthIcon />
+    }
+    if (type === 'boolean') {
+      return <CheckBoxIcon />
+    }
+    return <TextFieldsIcon />
+  }
+
+  return (
+    <Box>
+      <Alert sx={{ mt: 2, mb: 2 }} severity='info'>
+        This page is currently not fully implemented. More advanced schema options will be added in the future.
+      </Alert>
+      <Grid container spacing={2}>
+        <Grid item xs={12} sm={12} md={6}>
+          <Box>
+            <Typography sx={{ mb: 2 }} variant='h4'>
+              Configure Schema
+            </Typography>
+            <Paper
+              sx={{
+                p: 2,
+              }}
+            >
+              <Stack direction='row' spacing={2}>
+                <Tabs value={selectedStep} onChange={handleStepChange} variant='scrollable' scrollButtons='auto'>
+                  {steps.map((step) => (
+                    <Tab
+                      key={step.reference}
+                      value={step.reference}
+                      label={
+                        <span>
+                          {step.title}
+                          <IconButton
+                            size='small'
+                            component='span'
+                            sx={{ ml: 1 }}
+                            onClick={() => {
+                              deleteStep(step.reference)
+                            }}
+                          >
+                            <DeleteIcon />
+                          </IconButton>
+                        </span>
+                      }
+                      onClick={() => setSelectedStep(step.reference)}
+                    />
+                  ))}
+                </Tabs>
+
+                <Button
+                  color='primary'
+                  variant='text'
+                  startIcon={<AddIcon />}
+                  onClick={() => setNewStepDialogOpen(true)}
+                >
+                  Add step
+                </Button>
+              </Stack>
+
+              {steps.map((step) => (
+                <Box key={step.reference}>
+                  {step.reference === selectedStep && (
+                    <Box sx={{ pt: 2 }}>
+                      <Typography variant='caption'>Questions can be reordered by drag and drop</Typography>
+                      <DragDropContext onDragEnd={onQuestionDragEnd}>
+                        <Droppable droppableId='questionList'>
+                          {(droppableProvided) => (
```

Comment:
This is just typed as `any`, looks like we're missing the type definitions for this package. From what I can see `npm i -D @types/react-beautiful-dnd` should fix this.